File: EX2/PlantillasParaExamenFinal/Backend/SoftPlantillaBO/src/pe/edu/pucp/softplantilla/bo/PersonaBO.java
package pe.edu.pucp.softplantilla.bo;

import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softplantilla.dao.PersonaDAO;
import pe.edu.pucp.softplantilla.daoimp.PersonaDAOImp;
import pe.edu.pucp.softplantilla.model.Persona;
import pe.edu.pucp.softplantilla.model.Tipo;

public class PersonaBO {
	private PersonaDAO personaDAO;

	public PersonaBO() {
		personaDAO = new PersonaDAOImp();
	}

	public Integer insertar(String nombre, String apellido, Double altura, Tipo tipo, Date fechaNacimiento) {
		Persona persona = new Persona(nombre, apellido, altura, tipo, fechaNacimiento);
		return personaDAO.insertar(persona);
	}

	public Integer modificar(Integer idPersona, String nombre, String apellido, Double altura, Tipo tipo, Date fechaNacimiento) {
		Persona persona = new Persona(idPersona, nombre, apellido, altura, tipo, fechaNacimiento);
		return personaDAO.modificar(persona);
	}

	public Integer eliminar(Integer idPersona) {
		return personaDAO.eliminar(idPersona);
	}

	public ArrayList<Persona> listarTodas() {
		return personaDAO.listarTodas();
	}

	public Persona obtenerPorId(Integer idPersona) {
		return personaDAO.obtenerPorId(idPersona);
	}
}

File: EX2/PlantillasParaExamenFinal/Frontend/SoftPlantillaWS/SoftPlantillaBO/PersonaBO.cs
using SoftPlantillaBaseBO;
using SoftPlantillaBaseBO.ServicioWS;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftPlantillaBO
{
	public class PersonaBO : BaseBO
	{
		public int insertar(string nombre, string apellido, double altura, tipo tipo, DateTime fechaNacimiento)
		{
			return PersonaWS.insertarPersona(nombre, apellido, altura, tipo, fechaNacimiento);
		}

		public int modificar(int idPersona, string nombre, string apellido, double altura, tipo tipo, DateTime fechaNacimiento)
		{
			return PersonaWS.modificarPersona(idPersona, nombre, apellido, altura, tipo, fechaNacimiento);
		}

		public int eliminar(int idPersona)
		{
			return PersonaWS.eliminarPersona(idPersona);
		}

		public BindingList<persona> listarTodas()
		{
			persona[] personas = PersonaWS.listarPersonasTodas();
			if (personas == null)
				return new BindingList<persona>();
			else
				return new BindingList<persona>(personas);
		}

		public persona obtenerPorId(int idPersona)
		{
			return PersonaWS.obtenerPersonaPorId(idPersona);
		}
	}
}
